### DISCUSSION TITLE
alternative lock-ing strategie

### DISCUSSION THREAD

**Discussion by staabm** [deployphp/deployer#2400] [Help needed]
when deploying to multiple hosts things can easily get out of sync, because lock files get distributed across all hosts.
in the worst case, we get into a scenario where one of the hosts is locked while others are not.

I feel like it would be an improvement if we could get a single globl lock mechanism for all hosts.
either we can deploy or we can't.

this kind of scenario could be achieved e.g. using a redis based lock or similar. 
do you think it would make sense to use some kind of lock-ing abstraction in deployer instead of the file-based locking [we have right now](https://github.com/deployphp/deployer/blob/ff110c987d40250da0094dd82bdee775cbe8db64/recipe/deploy/lock.php)?

e.g. https://github.com/php-lock/lock looks like a good fit

**Comment by antonmedv** [comment]
Using lock files is the easiest lock implementation. Requiring some db for locks  will significantly increase complexity of majorities of users. But you definitely use any lock mechanism.

One simple solution for your case is:
task(“deploy:lock”)->select(“role=lock-host”)
